stdlib/src/builtin/uint.mojo

```diff
@@ -0,0 +1,86 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+"""Implements the UInt class.
+
+These are Mojo built-ins, so you don't need to import them.
+"""
+
+
+@lldb_formatter_wrapping_type
+@value
+@register_passable("trivial")
+struct UInt:
```

Comment:
**Question** Do you think we can get a lot of mileage/it would be sufficient to just have this be an alias to `UInt32` on 32-bit systems and `UInt64` otherwise?  @abduld just suggested this internally, which, sounds quite simple and would meet our needs if I understand correctly?

Reply:
The implementation would be 

```
alias UInt = UInt64 if bitwidthof[DType.index]() else UInt32
```

Reply:
Out of curiosity, is there any reason we're not doing this for Int then?

Reply:
I've also never been convinced that `Int` couldn't just be an alias so adding another seems unnecessary. If the user really needs an unsigned integer they can easily use one of the existing `Scalar` types?

Reply:
Int is really the same as Scalar[DType.index], but is implemented using the mlir index dialect rather than the compiler dialects (like pop) that is used by SIMD. I encourage one to try to replace the implementation of Int with an alias

Reply:
`Int` is special because it is the materialization target of `IntLiteral`. It also allows us to implement a bunch of other things that we couldn't do with `SIMD` because we lack proper conditional conformances. Turning `Int` into an alias is also not a simple task because the compiler depends on it. I'm +1 on trying to define `UInt` as an alias, but -1 on trying to change the `Int` implementation (at this point).

Reply:
I would strongly recommend not touching Into. It's special in a lot of ways

Reply:
Concretely, `Int` is backed by an `__mlir_type.index` because the `index` opset is more mature than the `pop` opset. The POP opset is also semantically different: we have `index.add_nsw` but `pop.add_nsw` doesn't make as much sense because it would only work for a subset of SIMD dtypes. Using an `index` has other benefits because it leads to terse IR, faster compile times, and also has lots of other load-bearing weirdness in MAX that would be very difficult to change.

Reply:
@JoeLoser Did you have the time to sync on the subject?